### DISCUSSION TITLE
curl --parallel: custom headers per each url

### DISCUSSION THREAD

**Discussion by AndriusCTR** [curl/curl#11489] [Q&A] [answered]
Is it possible to send different headers (or basically any command args) for each URL when using --parallel (-Z) option? Right now it assumes that each URL will require the same request headers... Which is often not the case. What if URLs require different headers, yet you want to utilize the parallel option for performance benefits?

I think it's not possible to achieve this currently, or is it?
Maybe having ability to include params / headers inside config.txt file for parallel procesing?

**Comment by dfandrich** [comment, marked as answer]
Does this not work using the --next option?

**Reply by AndriusCTR** [reply]
"If you’re doing parallel transfers (with [-Z](https://curl.haxx.se/docs/manpage.html#-Z)), curl will do URLs in parallel only until the --next separator."

https://daniel.haxx.se/blog/2020/03/02/curl-ootw-next/

Does it?

**Reply by bagder** [reply]
I believe I was wrong in that blog post. Have you tried doing it as @dfandrich suggests?
